include/message_filters/sync_policies/latest_time.h

```diff
@@ -0,0 +1,276 @@
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/**
+ * \brief Synchronizes up to 9 messages by their rates with upsampling via zero-order-hold.
+ *
+ * LatestTime policy synchronizes up to 9 incoming channels by the rates they are received.
+ * The callback with all the messages will be triggered whenever the fastest message is received.
+ * The slower messages will be repeated at the rate of the fastest message and will be updated
+ * whenever a new one is received. This is essentially an upsampling of slower messages using a
+ * zero-order hold (no interpolation).
+
+ * \section usage USAGE
+ * Example usage would be:
+\verbatim
+typedef LatestTime<sensor_msgs::CameraInfo, sensor_msgs::Image, sensor_msgs::Image> latest_policy;
+Synchronizer<latest_policy> sync_policies(latest_policy(1U), caminfo_sub, limage_sub, rimage_sub);
+sync_policies.registerCallback(callback);
+\endverbatim
+
+ * The callback is then of the form:
+\verbatim
+void callback(const sensor_msgs::CameraInfo::ConstPtr&, const sensor_msgs::Image::ConstPtr&, const sensor_msgs::Image::ConstPtr&);
+\endverbatim
+ *
+ */
+
+#ifndef MESSAGE_FILTERS__SYNC_LATEST_TIME_H_
+#define MESSAGE_FILTERS__SYNC_LATEST_TIME_H_
+
+#include <algorithm>
+#include <deque>
+#include <numeric>
+#include <string>
+#include <tuple>
+#include <vector>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include "message_filters/message_traits.h"
+#include "message_filters/null_types.h"
+#include "message_filters/signal9.h"
+#include "message_filters/synchronizer.h"
+
+
+namespace message_filters
+{
+namespace sync_policies
+{
+
+template<typename M0, typename M1,
+         typename M2 = NullType, typename M3 = NullType, typename M4 = NullType,
+         typename M5 = NullType, typename M6 = NullType, typename M7 = NullType,
+         typename M8 = NullType>
+struct LatestTime : public PolicyBase<M0, M1, M2, M3, M4, M5, M6, M7, M8>
+{
+  typedef Synchronizer<LatestTime> Sync;
+  typedef PolicyBase<M0, M1, M2, M3, M4, M5, M6, M7, M8> Super;
+  typedef typename Super::Messages Messages;
+  typedef typename Super::Signal Signal;
+  typedef typename Super::Events Events;
+  typedef typename Super::RealTypeCount RealTypeCount;
+
+  LatestTime(uint32_t)
```

Comment:
What's this unused argument for?  Can we just remove it?

Reply:
yep! removed